### DISCUSSION TITLE
Using bsky.network BGS/relay with the RepoCommitAdaptor

### DISCUSSION THREAD

**Discussion by ghost** [myConsciousness/atproto.dart#1001] [Q&A] [answered]
After the recent [change](https://github.com/bluesky-social/atproto/discussions/1832) with bsky.social migrating to multiple PDSs, how can I use the bsky.network's firehose instead of listening to the firehose of our account's PDS? I am following the tutorial "[How to Create Bluesky BOT using Dart and Firehose ](https://dev.to/shinyakato/how-to-create-bluesky-bot-using-dart-and-firehose-bll)", and wondering whether this method can be still used to get all events happening in all of bsky.social PDSs.

**Comment by myConsciousness** [comment, marked as answer]
Hi @Gusty9A !

Thanks for reaching out to me about this. I was not aware of this at all tbh.

Well you are right, now that the PDS federation has started we need to connect to the `bsky.network` to get Firehose for the whole bsky.social events.

I have released `atproto 0.8.1` and `bluesky 0.10.1` to handle this. Since these versions, we will always be connected to `bsky.network` by default for stream endpoints. Other than stream, it's `bsky.social` as before.

So you can get the whole bsky.social events as follows with the latest versions:

```dart
import 'package:bluesky/bluesky.dart' as bsky;

Future<void> main() async {
  // Now defaults to `bsky.network` for stream endpoint
  final bluesky = bsky.Bluesky.anonymous();

  final subscription = await bluesky.sync.subscribeRepoUpdates();

  subscription.data.stream.listen((event) {
    event.when(
      commit: bsky.RepoCommitAdaptor(
        //! Create events.
        onCreatePost: (data) => data.record,
        onCreateLike: print,

        //! Update events.
        onUpdateProfile: print,

        //! Delete events.
        onDeletePost: print,
      ).execute,
      handle: print,
      migrate: print,
      tombstone: print,
      info: print,
      unknown: print,
    );
  });
}
```

**Reply by ghost** [reply]
Thank you so much! I will test this out!
